test/components/elements/AutoCompleteMenuTest.js

```diff
@@ -0,0 +1,227 @@
+      })
+
+      it('the Autocomplete should have an onUpdateInput fx just like the fx we passed in', function () {
+          expect(this.output.props.children.props.onUpdateInput()).to.equal(mockProps.onUpdateInput())
+      })
+    })
+  })
+
+  describe('onNewRequest', function() {
+    const selectMethod = (input) => input
+    const clearMethod = () => console.log('')
+
+    context('when the input is valid', function() {
+      const dataSource = [{ name: 'Kermit' }, { name: 'Fozzie' }]
+      const input = 'k'
+      const clearSpy = chai.spy(clearMethod)
+      const selectSpy = chai.spy(selectMethod)
+
+      onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+      it('should call the clear method once', function() {
+        expect(clearSpy).to.have.been.called.once
+      })
+
+      it('should call the select method once', function() {
+        expect(selectSpy).to.have.been.called.once
+      })
+
+      describe('when the input has trailing spaces but matches an item in the data source', function() {
+        const dataSource = [{ name: 'Kermit' }, { name: 'Fozzie' }]
+        const input = 'Kermit   '
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should call the select method once', function () {
+          expect(selectSpy).to.have.been.called.once
+        })
+      })
+
+      describe('when the input is a different case but matches an item in the data source', function () {
+        const dataSource = [{ name: 'Kermit' }, { name: 'Fozzie' }]
+        const input = 'KERMIT'
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should call the select method once', function () {
+          expect(selectSpy).to.have.been.called.once
+        })
+      })
+
+      describe('when the input exactly matches an item in the data source', function() {
+        const dataSource = [{ name: 'Kermit' }, { name: 'Fozzie' }]
+        const input = 'Kermit'
+        const clearSpy = chai.spy(clearMethod)
+        const selectSpy = chai.spy(selectMethod)
+
+        onNewRequest(dataSource, selectSpy, clearSpy)(input)
+
+        it('should call the clear method once', function () {
+          expect(clearSpy).to.have.been.called.once
+        })
+
+        it('should call the select method once', function () {
+          expect(selectSpy).to.have.been.called.once
+        })
+      })
+
+    })
+
+    context('when the input is invalid', function() {
```

Comment:
I think this context is not _invalid_, but instead _when a clear autocomplete match can not be found_ and this context's sibling would be _when a clear autocomplete match can be found_.